src/core/Gsim.cc

```diff
@@ -480,8 +480,8 @@ void Gsim::MakeEvent(const G4Event* g4ev, DS::Root* ds) {
 
   // Get the PMT type for IDPMTs. Then in the loop,
   // increment numPE only when the PE is in an IDPMT.
-  // Map from PMT ID numbers to objects for use later in noise calculation
-  std::map<int, DS::MCPMT*> mcpmtObjects;
+  //Map ID-INDEX for later noise calculation
+  std::map<int, int> mcpmtObjects;
```

Comment:
The real problem is that we were defining pointers to the elements of the MCPMT std::vector. This is fine if you are not going to resize your vector. The issue arise when adding new PMTs due to dark hits, then the vector is relocated in memory and the pointers meaningless... I notice this problem in my simulation and the fix should work well.